src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1496,7 +1496,7 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Diag
 
                 case SymbolKind.Alias:
                     {
-                        var alias = symbol as AliasSymbol;
+                        var alias = (AliasSymbol)symbol;
                         symbol = alias.Target;
```

Comment:
@AlekseyTs said that this pattern is forbidden in the compiler coding guidelines, but I could not find that.